retrieval/target.go

```diff
@@ -361,6 +364,16 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	for samples := range t.ingestedSamples {
 		for _, s := range samples {
 			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
+			labels, err := Relabel(clientmodel.LabelSet(s.Metric), t.metricRelabelConfigs...)
```

Comment:
It wouldn't be correct to do it in `Relabel`, as targetmanager relies on it being a copy to not change `preRelabelLabels`. I've put in the caller for now. We'll want to reevalutate this interface in the future, as there's be another two uses of Relabel (alerting and remote storage) the latter of which is performance critical.